### ISSUE TITLE
ModuleNotFoundError: No module named 'shared_secret_authenticator'

### ISSUE THREAD

**Issue by ajay-actuary** [matrix-org/synapse#14963]
Description:
Upgrading matrixdotorg/synapse docker image from v.1.75 to v.1.76 results in the error: ModuleNotFoundError: No module named 'shared_secret_authenticator'. Possibly missing python module.

Steps to reproduce:
1. Install latest matrix image & mount shared_secret_authenticator.py volume
```
sudo docker run -d --name matrix --restart unless-stopped \
    -v /path/Matrix/:/data \
    -v /path/Matrix/shared_secret_authenticator.py:/usr/local/lib/python3.9/site-packages/shared_secret_authenticator.py \
    -p 8008:8008 \
    matrixdotorg/synapse:latest
```

Should return the error: ModuleNotFoundError: No module named 'shared_secret_authenticator'

2. Repeat the same by downgrading to v.1.75 & everything should work
```
sudo docker run -d --name matrix --restart unless-stopped \
    -v /path/Matrix/:/data \
    -v /path/Matrix/shared_secret_authenticator.py:/usr/local/lib/python3.9/site-packages/shared_secret_authenticator.py \
    -p 8008:8008 \
    matrixdotorg/synapse:v1.75.0
```

Homeserver
Local deployment with Docker

Synapse Version
v1.76.0

Installation Method
Docker (matrixdotorg/synapse)

Database
Postgresql

Workers
Single process

Platform
Docker engine running on Arch Linux

Configuration
No response

Anything else I can share on this? 

Thanks.

**Comment by Netfloex** [comment]
Thanks this fixed the issue for me, for anyone else wondering this was the required change:
```diff
 volumes:
-            - ./plugins/shared_secret_authenticator.py:/usr/local/lib/python3.9/site-packages/shared_secret_authenticator.py
+            - ./plugins/shared_secret_authenticator.py:/usr/local/lib/python3.11/site-packages/shared_secret_authenticator.py
 ```

**Comment by DMRobertson** [comment]
Yes, you've been bitten by https://github.com/matrix-org/synapse/issues/9944

It might be worth retroactively amending the upgrade notes to call this out

**Comment by ajay-actuary** [comment]
> Are you sure that the path is correct? The container image uses Python 3.11.
> 
> https://github.com/matrix-org/synapse/blob/bf82b56babc9e2cacba34f8878da3b3834914b3a/docker/Dockerfile#L23
> 
> https://github.com/matrix-org/synapse/blob/develop/CHANGES.md#updates-to-the-docker-image

Yes, that was the issue! Works if I change it to 3.11. Thanks!
```

sudo docker run -d --name matrix --restart unless-stopped \
    -v /path/Matrix/:/data \
    -v /path/Matrix/shared_secret_authenticator.py:/usr/local/lib/python3.9/site-packages/shared_secret_authenticator.py \
    -p 8008:8008 \
    matrixdotorg/synapse:latest
```

**Comment by clokep** [comment]
I think we can close this then. Thanks for confirming it worked for you!
